.github/workflows/release.yml

```diff
@@ -1,158 +1,206 @@
+        type: boolean
+        default: true
+      crate:
+        description: "The crate to release. (containerd-shim-wasmtime)"
+        type: string
+      version:
+        description: "The version of the crate to release. (e.g., 1.2.3)"
+        type: string
+
 concurrency:
-  group: release-${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  group: release-${{ github.workflow }}-${{ inputs.crate }}-${{ inputs.version }}
   cancel-in-progress: true
 
-on:
-  push:
-    tags:
-      - '**/v[0-9]+.[0-9]+.*'
-
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  parse:
-    runs-on: ubuntu-latest
-    name: Parse ref
+  pre-release:
+    name: pre-release checks
+    runs-on: "ubuntu-latest"
     outputs:
-      crate: ${{ steps.parse.outputs.crate }}
-      version: ${{ steps.parse.outputs.version }}
-      runtime: ${{ steps.parse.outputs.runtime }}
+      crate: ${{ inputs.crate }}
+      runtime: ${{ steps.runtime_sub.outputs.runtime }}
+      version: ${{ inputs.version }}
+      ### is_shim is a string, not a boolean, so use: is_shim == 'true'
+      is_shim: ${{ steps.runtime_sub.outputs.is_shim }}
     steps:
-      - uses: actions/checkout@v3
+      # - name: Fail if branch is not main
+      #   if: github.event_name == 'workflow_dispatch' && github.ref != 'refs/heads/main'
+      #   run: |
+      #     echo "This workflow should not be triggered with workflow_dispatch on a branch other than main"
+      #     exit 1
+
+      ### Determine the name of the runtime and if it is a binary release or crates.io
+      - name: substring runtime
+        id: runtime_sub
+        uses: actions/github-script@v7
         with:
-          ref: ${{ github.ref }}
-      - id: parse
-        name: Parse ref
-        shell: bash
-        run: ./scripts/parse_ref.sh ${{ github.ref }} >> ${GITHUB_OUTPUT}
+          script: |
+            const crate = '${{ inputs.crate }}';
+            let runtime = crate.replace(/^containerd-shim-/, '');
+            core.setOutput('runtime', runtime);
+            const non_shim_crates = ['wasm', 'wasm-test-modules', 'oci-tar-builder'];
+            core.setOutput('is_shim', !non_shim_crates.includes(runtime));
+      ### If we are releasing a crate rather than producing a bin, check for crates.io access
+      - name: Check crates.io ownership
+        if: ${{ steps.runtime_sub.outputs.is_shim != 'true' }}
+        run: |
+          cargo owner --list ${{ inputs.crate }} | grep github:containerd:runwasi-committers || \
+            cargo owner --add github:containerd:runwasi-committers ${{ inputs.crate }}
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_PUBLISH_TOKEN }}
+      - name: Update crate version and commit change to main
+        if: ${{ !inputs.dry_run }}
+        run: |
+          # replace the version inline in the Cargo.toml
+          sed -i '' -E 's/^version.+=.+".+"/version = "${{ inputs.version }}"/' crates/${{ inputs.crate }}/Cargo.toml
```

Comment:
Interested in what others think about this. I am afraid of typing the wrong thing into the input box in the GH UI and kicking off the wrong build here.  I think I'd prefer to do a PR to update the version then grab the latest version

Reply:
@Mossaka what say you?

Reply:
I can add some sanitation to make sure the build fails based on checking the crate names and the version regex. That could provide some basic sanity checks. However, it is not going to be a second set of human eyes, which may be the root of what you are seeking.

Reply:
>  However, it is not going to be a second set of human eyes, which may be the root of what you are seeking.

Yes but its probably not a huge deal

Reply:
Check out: https://github.com/containerd/runwasi/actions/runs/8483472409/job/23244657527.

I've added some sanity checks to the early part of the release.

Reply:
I tried to run this command on my ubuntu devbox but it gave an error 

```
sed: can't read s/^version.+=.+".+"/version = "0.5.1"/: No such file or directory
```

it seems like in Ubuntu, the command expects no space between `-i` and the backup extension `""`, so the correct command should be 

```bash
sed -i'' -E ...
```